source/cipher_openssl.c

```diff
@@ -127,17 +139,28 @@ size_t aws_cryptosdk_md_size(enum aws_cryptosdk_md_alg md_alg) {
 }
 
 int aws_cryptosdk_md_update(struct aws_cryptosdk_md_context *md_context, const void *buf, size_t length) {
+    AWS_PRECONDITION(aws_cryptosdk_md_context_is_valid(md_context));
+    AWS_PRECONDITION(buf);
+    AWS_PRECONDITION(AWS_MEM_IS_READABLE(buf, length));
```

Comment:
The previous assertion is subsumed by this one.  (unless we care that buf should be non-null even if the length is 0, in which case clearer to write ```buf && AWS_MEM_IS_READABLE(buf, length)```

Reply:
I've removed the previous assertion. Since the OpenSSL function that this calls returns immediately if the length is 0, it should be safe in this case even if buf is NULL. Unless @bdonlan disagrees.